### ISSUE TITLE
[BUG] Difference among distutils.sysconfig.get_python_inc() 

### ISSUE THREAD

**Issue by wqh17101** [pypa/setuptools#3605]
### setuptools version

65.3.0

### Python version

3.9.2

### OS

Linux

### Additional environment information

My env is in a docker.

### Description

Hi there, i find that setuptools 65.3.0 hack the distutils. I want to get the include dir of python
And i found that 

### before install setuptools 
```
python -c 'import distutils.sysconfig; print(distutils.sysconfig.get_python_inc())'
```
and get `/opt/aaa/python/python-3.9.2/include/python3.9`

### after install that
```
python -c 'import distutils.sysconfig; print(distutils.sysconfig.get_python_inc())'
```
and get `/devcloud/ws/sirFh/workspace/j_VWBQ8FGA/aaa_python/third_build/python/include/python3.9`

###
`/opt/aaa/python/python-3.9.2/include/python3.9` is the right anwser.
And  `/devcloud/ws/sirFh/workspace/j_VWBQ8FGA/aaa_python/third_build/python/include/python3.9` does not exist in my docker or even my machine

### Expected behavior

Get the right include dir

### How to Reproduce

import distutils.sysconfig; print(distutils.sysconfig.get_python_inc())

### Output

python -c 'import distutils.sysconfig; print(distutils.sysconfig.get_python_inc())'

**Comment by abravalheri** [comment]
Hi @wqh17101, could you try running this with https://github.com/pypa/distutils instead of the `distutils` that comes with the standard library?

The `distutils` library is marked for removal from the stdlib and is currently being maintained/receiving updates in https://github.com/pypa/distutils. When you install & import `setuptools`, it will automatically redirect the imports to use the latest updated version.

Regarding the reproducer, if we need to investigate the issue deeper we will need a bit more information than linux/docker. Is there any chance you can share a minimal Dockerfile that reproduces the error? You can find some tips on how to right helpful reproducers in [this post from stackoverflow](https://stackoverflow.com/help/minimal-reproducible-example).

**Comment by wqh17101** [comment]
So that you mean this is the bug of   https://github.com/pypa/distutils ?@abravalheri

**Comment by abravalheri** [comment]
> So that you mean this is the bug of https://github.com/pypa/distutils ?

I don't think we can be categorical with this without a proper reproducer. Could you have a look on the 2 items I proposed in the previous comment?

> could you try running this with https://github.com/pypa/distutils instead of the distutils that comes with the standard library?

Have you tried this one? What is the outcome?

> Is there any chance you can share a minimal Dockerfile that reproduces the error? You can find some tips on how to right helpful reproducers in [this post from stackoverflow](https://stackoverflow.com/help/minimal-reproducible-example).

Could you please share with us a reproducer?

**Comment by wqh17101** [comment]
> > could you try running this with https://github.com/pypa/distutils instead of the distutils that comes with the standard library?
> 
> Have you tried this one? What is the outcome?

How to install this one?

Sorry,Hard to make a reproducer for me.

**Comment by abravalheri** [comment]
> How to install this one?

If you clone https://github.com/pypa/distutils and run the tests from within the folder, that should do the trick.

> Sorry,Hard to make a reproducer for me.

This makes it really difficult to investigate/help.

**Comment by wqh17101** [comment]
> > How to install this one?
> 
> If you clone https://github.com/pypa/distutils and run the experiment from within the folder, that should do the trick.

get the same result as setuptools,`/devcloud/ws/sirFh/workspace/j_VWBQ8FGA/aaa_python/third_build/python/include/python3.9`





> > Sorry,Hard to make a reproducer for me.
> 
> This makes it really difficult to investigate/help.

Maybe you can test in any docker env?

**Comment by abravalheri** [comment]
> Maybe you can test in any docker env?

I cannot observe the described behaviour in the following experiment:

```bash
> docker run --rm -it python:3.9.2 /bin/bash

python -c 'import distutils.sysconfig; print(distutils.sysconfig.get_python_inc())'
# /usr/local/include/python3.9

python -m pip install -U pip setuptools

python -c 'import distutils.sysconfig; print(distutils.sysconfig.get_python_inc())'
# /usr/local/include/python3.9
```

**Comment by wqh17101** [comment]
@abravalheri I found the reason. https://github.com/pypa/distutils/issues/179

**Comment by zjp** [comment]
https://github.com/pypa/distutils/issues/178#issuecomment-1257270886

I opened a similar issue in the distutils repository. I have a reproducible case with ChimeraX.

**Comment by zjp** [comment]
If you prefer not to install ChimeraX, you can pull our docker container ``rbvi/chimerax:latest``. 

```
[root@3be7e637a87b ChimeraX]# ChimeraX.app/bin/ChimeraX -c 'from distutils import sysconfig; print(sysconfig.get_python_inc())'
/ChimeraX/ChimeraX.app/include/python3.9

[root@3be7e637a87b ChimeraX]# ChimeraX.app/bin/ChimeraX -m pip install -U setuptools
Requirement already satisfied: setuptools in ./ChimeraX.app/lib/python3.9/site-packages (65.1.1)
Collecting setuptools
  Downloading setuptools-65.5.0-py3-none-any.whl (1.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 12.9 MB/s eta 0:00:00
Installing collected packages: setuptools
  Attempting uninstall: setuptools
    Found existing installation: setuptools 65.1.1
    Uninstalling setuptools-65.1.1:
      Successfully uninstalled setuptools-65.1.1
Successfully installed setuptools-65.5.0
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

[notice] A new release of pip available: 22.2.2 -> 22.3
[notice] To update, run: /ChimeraX/ChimeraX.app/bin/ChimeraX -m pip install --upgrade pip

[root@3be7e637a87b ChimeraX]# ChimeraX.app/bin/ChimeraX -c 'from distutils import sysconfig; print(sysconfig.get_python_inc())'
/ChimeraX/build/include/python3.9
```

As you can see, the header locations change from the bundled Python's location to wherever they were when that Python was built.
